### PR TITLE
feat(oaf-discriminators): support discriminators for oneof/anyof

### LINKED ISSUE #37: Add discriminators handling for oneOf/anyOf cases

### PR DESCRIPTION
**Pull request by asadali214**
This PR adds support for discriminators in oneOf/anyOf cases using existsing oneOf/anyOf templates and discriminator substitutions.
It also contains new unit tests to verify this feature

Closes #37

### CODE REVIEW

File: src/TypeCombination.php [hunk 1]
```diff
@@ -63,9 +78,67 @@ class TypeCombination
     private function __construct($format, $groupName, $types, $deserializers)
     {
         $this->_format = $format;
-        $this->_groupName = $groupName;
+        $this->_groupName= $groupName;
```
Comment (sufyankhanrao) [inline src/TypeCombination.php:81]: Nits, Non-blocker: Removing space before the assignment operator should be reverted here.

Comment (asadali214) [reply src/TypeCombination.php]: fixed

File: src/OneOfValidationException.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * OneOf Validation Exception.
+ *
+ * @category Apimatic
+ * @package  JsonMapper
+ * @author   Asad Ali <asad.ali@apimatic.io>
+ * @license  OSL-3.0 http://opensource.org/licenses/osl-3.0
+ * @link     https://www.apimatic.io/
+ */
+class OneOfValidationException extends JsonMapperException
```
Comment (sufyankhanrao) [inline src/OneOfValidationException.php:25]: This exception class should also contain the case when no matching case is found while validating the OneOf types.

Comment (asadali214) [reply src/OneOfValidationException.php:25]: I have added the suggested exception and updated its related tests

**Review by sufyankhanrao — APPROVED** [review]
LGTM

**Pull request merged**